src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+  });
+
+  test(`should render row header with icon node as children`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableSpanComponent />);
+
+    await expect(
+      flatTableRowHeader(page).nth(0).locator("span")
+    ).toHaveAttribute("data-component", "icon");
+  });
+
+  test(`should render row header with string as children`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableSpanComponent />);
+
+    await expect(flatTableRowHeader(page).nth(0)).toHaveText("Code - Sticky ");
+  });
+
+  test(`should render cell with string as children`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableTruncateBgComponent />);
+
+    await expect(flatTableCell(page, 0)).toHaveText("John Doe");
+    await expect(flatTableCell(page, 1)).toHaveText("London");
+  });
+
+  test(`should render cell with div node as children`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableTruncateBgComponent />);
+
+    await expect(flatTableCell(page, 2).locator("input")).toHaveCount(1);
+  });
+
+  test(`should render checkbox with ariaLabelledBy`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(<FlatTableColorRowSelectableComponent />);
+
+    await expect(
+      flatTableHeaderCells(page).nth(0).locator("input")
+    ).toHaveAttribute("aria-labelledby", CHARACTERS.STANDARD);
+  });
+
+  test(`should render with sticky header`, async ({ mount, page }) => {
+    await mount(
+      <div style={{ height: "150px" }}>
+        <FlatTableComponent hasStickyHead />
+      </div>
+    );
+
+    await expect(flatTable(page)).toHaveCount(1);
+    await expect(flatTable(page).locator("thead")).toHaveCSS(
+      "position",
+      "sticky"
+    );
+
+    for (let i = 0; i < 5; i++) {
+      if (i === 3 || i === 4) {
+        await expect(flatTableBodyRowByPosition(page, i)).not.toBeInViewport();
+      } else {
+        await expect(flatTableBodyRowByPosition(page, i)).toBeInViewport();
+      }
+    }
+
+    const wrapperElement = flatTableWrapper(page);
+    await wrapperElement.evaluate((wrapper) => wrapper.scrollBy(0, 150));
+
+    for (let i = 0; i < 5; i++) {
+      if (i === 0 || i === 1) {
+        await expect(flatTableBodyRowByPosition(page, i)).not.toBeInViewport();
+      } else {
+        await expect(flatTableBodyRowByPosition(page, i)).toBeInViewport();
+      }
+    }
+  });
+
+  test(`should render with sticky header and multiple rows`, async ({
+    mount,
+    page,
+  }) => {
+    await mount(
+      <div style={{ height: "150px" }}>
+        <FlatTableSpanComponent hasStickyHead />
+      </div>
+    );
+
+    await expect(
+      flatTableHeaderRowByPosition(page, 0).locator("th").first()
+    ).toHaveCSS("top", "0px");
+    await expect(
+      flatTableHeaderRowByPosition(page, 1).locator("th").first()
+    ).toHaveCSS("top", "40px");
+  });
+
+  test(`should render with sticky footer`, async ({ mount, page }) => {
```

Comment:
Comment (non-blocking): these type of tests are quite complex (for loops with if/ else etc), I think these should also be reviewed in the future work done after this epic. We may be able to simplify it by asserting the hight of the table scrollable container and that in the first interaction the last row is not in viewport but the first is and then after the second interaction that the first is not visible but the last is etc.


```
test(`should render with sticky footer`, async ({ mount, page }) => {
    await mount(<FlatTableFooterComponent hasStickyFooter />);

    await expect(flatTablePager(page).locator("..")).toHaveCSS(
      "position",
      "sticky"
    );

    const wrapperElement = flatTableWrapper(page);
    const height = await wrapperElement.evaluate((el) =>
    window
      .getComputedStyle(el.parentElement as HTMLElement)
      .getPropertyValue("height")
  )
    expect(height).toBe("200px")

    await expect(flatTableBodyRowByPosition(page, 4)).not.toBeInViewport();
    await expect(flatTableBodyRowByPosition(page, 0)).toBeInViewport();

    await wrapperElement.evaluate((wrapper) => wrapper.scrollBy(1366, 150));
    const pagerParent = flatTablePager(page).locator("..");
    await pagerParent.scrollIntoViewIfNeeded();

    await expect(flatTableHeaderCells(page).first()).not.toBeInViewport();
    await expect(pagerParent).toBeInViewport();

    await expect(flatTableBodyRowByPosition(page, 0)).not.toBeInViewport();
    await expect(flatTableBodyRowByPosition(page, 4)).toBeInViewport();
  });